### ISSUE TITLE
Rebind action assumes default input always exists - Can't rebind any inputs to vJoy - "A needed vJoy device is not accessible"

### ISSUE THREAD

**Issue by mrkline** [WhiteMagic/JoystickGremlin#189]
I've been using Joystick Gremlin to map joystick inputs to macros with no issue, but I get the following when I try to remap an input to a vJoy press:

![error](https://user-images.githubusercontent.com/493463/52525818-190aca00-2c64-11e9-8b65-1b2c0473092e.png)

Unfortunately, the log doesn't provide any more detailed information:
```
2019-02-09 12:14:40      ERROR 'vJoy device 1 input 1 is not available'
2019-02-09 12:14:41      ERROR 'vJoy device 1 input 1 is not available'
```

The obvious first guess would be that vJoy isn't enabled or something, but things seem fine on that end:
![vJoy running](https://user-images.githubusercontent.com/493463/52525846-91718b00-2c64-11e9-9f53-0f418c1108f3.png)

And earlier in the log, Joystick Gremlin seems to find vJoy just fine:
```
2019-02-09 12:05:04      DEBUG Added: name=vJoy Device windows_id=0 hardware_id=305446573
2019-02-09 12:05:04      DEBUG vJoy windows id 0: (8, 8, 0)
2019-02-09 12:05:04      DEBUG vjoy id 1: (8, 8, 0) - MATCH
```

**Comment by WhiteMagic** [comment]
I think in this case the issue stems from Gremlin trying to select / enumerate the hats on the vJoy device which are set to 0 in your vjoy configuration. From a quick test adding a remap to a physical hat causes this problem as Gremlin tries to highlight the first hat by default for physical hats. Though because no hat exists that fails and shows the error. As a result of the error the ui isn't fully initialized and you don't get the "virtual button" tab, if you select a button from the drop down the missing tab should appear.

The error is misleading in this case as the device is accessible but simply doesn't have the expected input available. In the current dev version this actually causes other issues but in either case the assumption of the input existing should be fixed.

**Comment by mrkline** [comment]
Adding some hats to the vJoy device seems to be a workaround. Thanks!

**Comment by WhiteMagic** [comment]
This should have been fixed with 89ce4ba0a4edfc90bf92a3382d01f7c8c9bbff95
